robot/web/web.go

```diff
@@ -498,6 +503,68 @@ func (svc *webService) runWeb(ctx context.Context, options weboptions.Options) (
 	return err
 }
 
+// Namer is used to get a resource name from incoming requests for countingfor request. Requests for
+// resources are expected to be a gRPC object that includes a `GetName` method.
+type Namer interface {
+	GetName() string
+}
+
+// RequestCounter maps string keys to atomic ints that get bumped on every incoming gRPC request for
+// components.
+type RequestCounter struct {
+	counts sync.Map
+}
+
+// UnaryInterceptor returns an incoming server interceptor that will pull method information and
+// optionally resource information to bump the request counters.
+func (mc *RequestCounter) UnaryInterceptor(
+	ctx context.Context, req any, info *googlegrpc.UnaryServerInfo, handler googlegrpc.UnaryHandler,
+) (resp any, err error) {
+	// Handle `info.FullMethod` values such as:
+	// - `/viam.component.motor.v1.MotorService/IsMoving`
+	// - `/viam.robot.v1.RobotService/SendSessionHeartbeat`
+	//
+	// Only count component APIs, for now.
+	var apiMethod string
+	switch {
+	case strings.HasPrefix(info.FullMethod, "/viam.component."):
```

Comment:
is there a downside in adding services as well?

Reply:
Nope -- just didn't want to deal with double checking all of the kinds of method names to do string parsing on.